src/lib/y2network/sysconfig/interfaces_writer.rb

```diff
@@ -54,10 +57,40 @@ def udev_rule_for(iface)
         end
       end
 
-      # Refreshes udev service
-      def update_udevd
+      # Renames interfaces and refresh the udev service
+      #
+      # @param interfaces [InterfaceCollection] Interfaces
+      def update_udevd(interfaces)
+        udev_rules = interfaces.map { |i| udev_rule_for(i) }.compact
+        Y2Network::UdevRule.write(udev_rules)
+
         Yast::Execute.on_target("/usr/bin/udevadm", "control", "--reload")
         Yast::Execute.on_target("/usr/bin/udevadm", "trigger", "--subsystem-match=net", "--action=add")
+        # wait so that ifcfgs written in NetworkInterfaces are newer
+        # (1-second-wise) than netcontrol status files,
+        # and rcnetwork reload actually works (bnc#749365)
+        Yast::Execute.on_target("/usr/bin/udevadm", "settle")
+        sleep(1)
+      end
+
+      # Cleans and shutdowns renamed interfaces
+      #
+      # @param interfaces [InterfacesCollection] Interfaces
+      def clean_up_old_interfaces(interfaces)
+        interfaces.to_a.select(&:old_name).each do |iface|
```

Comment:
what happens if I switch names. So I have eth0 and eth1 and I change it to eth1 and eth0? This looks like it it is removed  and shutted down, not?

Reply:
Yes, that's it. And you need to do so in order to rename the interfaces later (if the interface is still up, it won't be renamed).